test/lib/epochs.test.js

```diff
@@ -256,3 +260,86 @@ test('lib/epochs (tieBreak)', async t => {
 
   t.end()
 })
+
+test('lib/epochs (getPreferredEpoch - 4.4. same membership)', async t => {
+  const run = Run(t)
+
+  const peers = [Server(), Server(), Server()]
+  t.teardown(() => peers.forEach(peer => peer.close(true)))
+
+  const [alice, bob, oscar] = peers
+  const [aliceId, bobId, oscarId] = await getRootIds(peers)
+  await run(
+    'start tribes',
+    Promise.all(peers.map(peer => peer.tribes2.start())),
+  )
+
+  const group = await run('alice creates a group', alice.tribes2.create({}))
+
+  await run(
+    '(sync dm feeds)',
+    Promise.all([replicate(alice, bob), replicate(alice, oscar)])
+  )
+
+  await run(
+    'alice invites bob, oscar',
+    alice.tribes2.addMembers(group.id, [bobId, oscarId], {})
+  )
+
+  await run(
+    '(sync invites)',
+    Promise.all([replicate(alice, bob), replicate(alice, oscar)])
+  )
+
+  await run(
+    'others accept invites',
+    Promise.all([
+      bob.tribes2.acceptInvite(group.id),
+      oscar.tribes2.acceptInvite(group.id),
+    ])
+  )
+
+  await run(
+    'bob and oscar both exclude alice (mutiny, fork)!',
+    Promise.all([
+      bob.tribes2.excludeMembers(group.id, [aliceId], {}),
+      oscar.tribes2.excludeMembers(group.id, [aliceId], {})
+    ])
+  )
+
+  await run('(sync exclusion)', replicate(bob, oscar))
+
+  const epochs = await Epochs(oscar).getEpochs(group.id)
+    .then(epochs => epochs.filter(epoch => epoch.author != aliceId))
+  const preferredEpoch = Epochs({}).tieBreak(epochs)
+
+  t.deepEqual(
+    await Epochs(oscar).getPreferredEpoch(group.id),
+    preferredEpoch,
+    'getPreferredEpoch'
+  )
+
+  // TODO need to test epochs > 2
+
+  t.end()
+})
+
+test.skip('lib/epochs (getPreferredEpoch - 4.5. subset membership)', async t => {
+  // the choice is the epoch which is a subset of all the others
+
+  t.end()
+})
+
+test.skip('lib/epochs (getPreferredEpoch - 4.6. overlapping membership)', async t => {
+  // there is no preferred epoch, you will need to choose choose a winner
+  // and exclude some members
+  // (not sure what the API should be here)
```

Comment:
maybe, return an epoch that the user can use for now (in case they're planning to post a message this second), but also include a soft error that stuff needs to be resolved asap?